NineChronicles.Headless/GraphTypes/StateQuery.cs

```diff
@@ -320,6 +321,86 @@ public StateQuery()
                     return null;
                 }
             );
+
+            Field<RaiderStateType>(
+                name: "raiderState",
+                description: "world boss season user information.",
+                arguments: new QueryArguments(
+                    new QueryArgument<NonNullGraphType<AddressType>>
+                    {
+                        Name = "raiderAddress",
+                        Description = "address of world boss season."
+                    }
+                ),
+                resolve: context =>
+                {
+                    var raiderAddress = context.GetArgument<Address>("raiderAddress");
+                    if (context.Source.GetState(raiderAddress) is List list)
+                    {
+                        return new RaiderState(list);
+                    }
+
+                    return null;
+                }
+            );
+
+            Field<NonNullGraphType<IntGraphType>>(
+                "raidId",
+                description: "world boss season id by block index.",
+                arguments: new QueryArguments(new QueryArgument<NonNullGraphType<LongGraphType>>
+                {
+                    Name = "blockIndex"
+                }),
+                resolve: context =>
+                {
+                    var blockIndex = context.GetArgument<long>("blockIndex");
+                    var sheet = new WorldBossListSheet();
+                    var address = Addresses.GetSheetAddress<WorldBossListSheet>();
+                    if (context.Source.GetState(address) is Text text)
+                    {
+                        sheet.Set(text);
+                    }
+
+                    return sheet.FindRaidIdByBlockIndex(blockIndex);
```

Comment:
Though I'm not an expert of the world boss feature, is it okay to find raid id even if there is no sheet? (empty sheet).

When I saw the `FindRaidIdByBlockIndex(long)` extension method, it seems failed when there is no elements:

```csharp
        public static WorldBossListSheet.Row FindRowByBlockIndex(this WorldBossListSheet sheet,
            long blockIndex)
        {
            return sheet.OrderedList
                .First(r =>
                    r.StartedBlockIndex <= blockIndex &&
                    blockIndex <= r.EndedBlockIndex
                );
        }
```

Maybe the type of `raidId` should be nullable `IntGraphType`.

Reply:
`FindRowByBlockIndex` throw exception when sheet is empty. so the type seems to be non-nullable. I think it's like `Linq.First` and `Linq.FirstorDefault`.

Reply:
Oops you're right. I'll resolve this comment.

---

I understood that with the below example:

```csv
id,boss_id,started_block_index,ended_block_index,fee,ticket_price,additional_ticket_price,max_purchase_count
1,900001,0,100,1,5,5,40
2,900001,200,300,1,5,5,40
```

Throwing the error from `Linq.First` means there is no boss for the block index. For example, it will also throw error when the block index is more than 300 with the sheet. If the sheet state doesn't exist, it works like the example.